### PR TITLE
Adding quillt pkgdown site built by GHA

### PR DESCRIPTION
**Pull request by apreshill**
👋 @rich-iannone 

Draft for now so we have something to talk about 😄

### CODE REVIEW

**Comment by apreshill** [comment]
Update: I have the Github Action working to build then deploy to Netlify. 

Draft url: https://blastula-pkg.netlify.app/

We'll need to use the GH action in this PR to deploy to Netlify (no branches), so that ultimately we can add the package here: https://pkgs.rstudio.com/

And use Netlify rewrites to get you the url: https://pkgs.rstudio.com/blastula

I did have a few build problems with 2 undocumented functions- I added them to get the build working, but perhaps not in the right place?
https://github.com/apreshill/blastula/blob/3b30528da0ca64594bea70bfda3355c04941f21e/_pkgdown.yml#L40-L41

File: .github/workflows/pkgdown.yaml [hunk 1]
```diff
@@ -1,50 +1,70 @@
 on:
   push:
-    branches: master
+    branches: quillt
+  workflow_run:
+    workflows: ["R-CMD-check"]
+    branches: [main,master]
+    types: completed
```
Comment (cderv) [inline .github/workflows/pkgdown.yaml:7]:
```suggestion
```

This should be removed. See last edition to the worflow I have made in https://github.com/rstudio/rmarkdown/blob/master/.github/workflows/pkgdown.yaml

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -1,50 +1,70 @@
...
 
       - uses: r-lib/actions/setup-pandoc@master
 
-      - name: Query dependencies
+      - name: Install pak and query dependencies
         run: |
-          install.packages('remotes')
-          saveRDS(remotes::dev_package_deps(dependencies = TRUE), ".github/depends.Rds", version = 2)
+          install.packages("pak", repos = "https://r-lib.github.io/p/pak/dev/")
+          saveRDS(pak::pkg_deps("local::.", dependencies = TRUE), ".github/r-depends.rds")
```
Comment (cderv) [inline .github/workflows/pkgdown.yaml:30]:
```suggestion
          saveRDS(pak::pkg_deps(".", dependencies = TRUE), ".github/r-depends.rds")
```
Pak supports this simpler syntax now.

File: .github/workflows/pkgdown.yaml [hunk 3]
```diff
@@ -1,50 +1,70 @@
...
+      - name: Deploy to Netlify
+        # only deploy on the fork for now
+        if: github.repository_owner == 'apreshill'
+        # NETLIFY_AUTH_TOKEN added in the repo's secrets
+        env:
+          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+          NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+        uses: netlify/actions/cli@master
+        with:
+          args: deploy --prod --dir docs
```
Comment (cderv) [inline .github/workflows/pkgdown.yaml:70]: Once ready to merge, we can change this to allow deploy preview on PR for example. See 
https://github.com/rstudio/rmarkdown/blob/master/.github/workflows/pkgdown.yaml

File: README.Rmd [hunk 4]
```diff
@@ -17,7 +17,7 @@ knitr::opts_chunk$set(
 library(blastula)
 ```
 
-# blastula <a href='http://rstudio.github.io/blastula/'><img src="man/figures/logo.svg" align="right" height="250px" /></a>
+# blastula <a href='https://rstudio.github.io/blastula'><img src='man/figures/logo.svg' align="right" height="139" /></a>
 
 [![CRAN status](https://www.r-pkg.org/badges/version/blastula)](https://CRAN.R-project.org/package=blastula)
 [![Travis-CI Build Status](https://travis-ci.org/rich-iannone/blastula.svg?branch=master)](https://travis-ci.org/rich-iannone/blastula)
```
Comment (cderv) [inline README.Rmd:23]: Maybe switch to Github action ? 
```suggestion
[![R-CMD-check](https://github.com/rstudio/blastula/workflows/R-CMD-check/badge.svg)](https://github.com/rstudio/blastula/actions)
```

**Review by cderv — COMMENTED** [review]
A few comments based on what we have done for others websites

* Maybe organize the README like the others, with a usage section containing the two demos currently in the README ? https://pkgs.rstudio.com/rmarkdown/

* For the examples tab and or Get Started vignnetes (which we don't have here yet), we maybe could link to some content for use blastula with RStudio Connect ? Or have a specific vignettes about that. Here are using links: 
    - How-to: 
        - https://solutions.rstudio.com/post/rstudio-connect-custom-emails-with-blastula/
    - Examples: 
        - https://solutions.rstudio.com/examples/blastula-overview/
        - https://solutions.rstudio.com/examples/blastula/conditional-example/
        - https://solutions.rstudio.com/examples/scheduling/scheduling/

Also we have a recipe in the cookbook if we want to link it somewhere: https://bookdown.org/yihui/rmarkdown-cookbook/blastula-email.html

@rich-iannone Happy to help with the GHA workflow change if you need as I did the ones for the other packages.

**Comment by apreshill** [comment]
Thank you @cderv! @rich-iannone we need to set up a time to scheme :)

**Comment by rich-iannone** [comment]
Thank you so much for all of this! @apreshill let's definitely talk about what else can be done here.

File: .github/workflows/pkgdown.yaml [hunk 5]
```diff
@@ -1,50 +1,66 @@
...
-          remotes::install_github("tidyverse/tidytemplate")
+          pak::local_system_requirements(execute = TRUE)
+          pak::pkg_system_requirements("pkgdown", execute = TRUE)
+        shell: Rscript {0}
+
+      - name: Install package dependencies
+        run: |
+          pak::local_install_dev_deps(upgrade = TRUE)
+          pak::pkg_install("r-lib/pkgdown")
+          pak::pkg_install("apreshill/quillt")
```
Comment (cderv) [inline .github/workflows/pkgdown.yaml:47]:
```suggestion
          pak::pkg_install("rstudio/quillt")
```

**quillt** has moved to rstudio/quillt

**Comment by rich-iannone** [comment]
Fixes: https://github.com/rstudio/blastula/issues/196

File: .github/workflows/pkgdown.yaml [hunk 6]
```diff
@@ -1,50 +1,83 @@
...
-
-      - name: Deploy package
-        run: pkgdown::deploy_to_branch(new_process = FALSE)
+      - name: Build pkgdown site
+        run: pkgdown::build_site(new_process = FALSE)
         shell: Rscript {0}
+
+      - name: Deploy to Netlify
+        # temp - only deploy on the fork for now, rm if
+        if: github.repository_owner == 'apreshill'
```
Comment (cderv) [inline .github/workflows/pkgdown.yaml:66]: Reminder: Don't forget to change this before merging.

File: .github/workflows/pkgdown.yaml [hunk 7]
```diff
@@ -1,50 +1,83 @@
...
+        id: netlify-deploy
+        uses: nwtgck/actions-netlify@v1.1
+        # NETLIFY_AUTH_TOKEN added in the repo's secrets
+        with:
+          publish-dir: 'docs'
+          # temp - change production-branch
+          production-branch: quillt
```
Comment (cderv) [inline .github/workflows/pkgdown.yaml:73]: Reminder: Don't forget to change this before merging.

**Comment by apreshill** [comment]
@rich-iannone for consistency, can you:

1. change "Introduction" to "Get started"?
2. Move examples before articles?

Looks great!

**Comment by apreshill** [comment]
Shall we launch this thing? We'll need to update the workflow, but this URL is already working: https://pkgs.rstudio.com/blastula/

**Comment by rich-iannone** [comment]
YES! Let's make it happen.

**Review by rich-iannone — DISMISSED** [review]

**Review by rich-iannone — APPROVED** [review]

**Pull request merged**